lib/amazon_ssa_support/rolling_s3_logger.rb

```diff
@@ -0,0 +1,79 @@
+require "logger"
+require "time"
+
+module AmazonSsaSupport
+  class RollingS3Logger < Logger
+    def initialize(logdev, aws_args: nil, maxsize: 1048576, log_to_stderr: nil, **options)
+      extractor_id = aws_args.try(:[], :extractor_id)
+      options[:formatter] ||= lambda do |severity, datetime, progname, msg|
+        "#{severity} [#{datetime.utc.iso8601(6)}#{extractor_id && " #{extractor_id}"}] -- #{progname}: #{msg}\n"
+      end
+
+      @stderr_logger = Logger.new(STDERR) if log_to_stderr
+
+      super(logdev, 1, maxsize, options)
```

Comment:
The options will set the level, progname and formatter by calling the respective setters. Since we overrode the setters below, this is how we mirror the logging to the stderr_logger if the option is set.